src/test/java/TestUtils.java

```diff
@@ -110,10 +132,27 @@ public static void setSearchServletRequestParameters(HttpServletRequest request,
 
     return analytics;
   }
-  /* * Removes the unique id property from a receipt entity JSON string, leaving only the receipt
+
+  /**
+   * Removes the unique id property from a receipt entity JSON string, leaving only the receipt
    * properties.
    */
   public static String extractProperties(String json) {
     return json.substring(json.indexOf("propertyMap"));
   }
+
+  /**
+   * Checks receipt ids in the returned page all match the ids in the expected page.
+   * @return true if all expectedPage receipt ids are found in returnedPage, else false.
+   */
+  public static boolean checkIdsMatch(ImmutableList<Entity> expectedPage, Receipt[] returnedPage) {
+    Supplier<Stream<Long>> ids = () -> Arrays.stream(returnedPage).map(Receipt::getId);
+
+    for (Entity expectedEntity : expectedPage) {
```

Comment:
Maybe also check the size matches so that returned doesn't contain more than expected?

Reply:
I assert for that in lines 341 and 378 in SearchServletTest. So by the time this method is called the sizes are guaranteed to be the same already.

Reply:
Would it make sense to put that check in this method since they are tightly coupled (checkIdsMatch() can't validate all id's match without ensuring the size is the same)?